File: examples/clients/sdk/identity.mjs
import { createClient } from "@coinrithm/sdk";

const apiKey = process.env.COINRITHM_API_KEY;
if (!apiKey) throw new Error("Set COINRITHM_API_KEY to your personal key.");
const client = createClient({
  apiKey,
  baseUrl: process.env.COINRITHM_BASE_URL || "https://api.coinrithm.com",
});
const { data, response } = await client.GET("/api/agent/me", {
  signal: AbortSignal.timeout(30_000),
});
if (!response.ok || !data)
  throw new Error(`API request failed: HTTP ${response.status}`);
// Account identity and scopes; never print the API key itself.
console.log(JSON.stringify(data, null, 2));

File: examples/clients/python/identity.py
import json
import os

import httpx
from coinrithm_sdk import AuthenticatedClient
from coinrithm_sdk.api.identity import whoami

api_key = os.getenv("COINRITHM_API_KEY")
if not api_key:
    raise RuntimeError("Set COINRITHM_API_KEY to your personal key.")
with AuthenticatedClient(
    base_url=os.getenv("COINRITHM_BASE_URL", "https://api.coinrithm.com"),
    token=api_key,
    timeout=httpx.Timeout(30.0),
) as client:
    response = whoami.sync_detailed(client=client)
    if response.status_code != 200 or response.parsed is None:
        raise RuntimeError(f"API request failed: HTTP {response.status_code}")
    # Account identity and scopes; never print the API key itself.
    print(json.dumps(response.parsed.to_dict(), indent=2))
